docs/advanced/prompt_suggestions.md

```diff
@@ -0,0 +1,237 @@
+YANG repo: [YangModels/yang](https://github.com/YangModels/yang)
+
+Clone the `YangModels/yang` repo and change into the main directory of the repo:
+
+```
+git clone https://github.com/YangModels/yang
+cd yang/vendor
+```
+
+##### IOS-XR
+The IOS-XR native YANG models are disaggregated and spread all over the place. Although its technically possible to load them all in one go, this approach will produce a lot of top-level modules making the navigation quite hard.
+
+An easier and cleaner approach would be to find the relevant module(s) and load them separately or in small batches. For example here we load BGP config and operational models together:
+
+```
+gnmic --file vendor/cisco/xr/721/Cisco-IOS-XR-um-router-bgp-cfg.yang \
+      --file vendor/cisco/xr/721/Cisco-IOS-XR-ipv4-bgp-oper.yang \
+      --dir standard/ietf \
+      prompt
+```
+
+!!! note
+    We needed to include the `ietf/` directory by means of the `--dir` flag, since the Cisco's native modules rely on the IETF modules and these modules are not in the same directory as the BGP modules.
+
+##### NX-OS
+Cisco NX-OS native modules, on the other hand, are aggregated in a single file, here is how you can generate the suggestions from it:
+
+```
+gnmic --file vendor/cisco/xr/721/Cisco-IOS-XR-um-router-bgp-cfg.yang \
+      --dir standard/ietf \
+      prompt
+```
+
+#### Juniper
+YANG repo: [Juniper/yang](https://github.com/Juniper/yang)
+
+Clone the Juniper YANG repository and change into the release directory:
+
+```
+git clone https://github.com/Juniper/yang
+cd yang/20.3/20.3R1
+```
+
+Start `gnmic` and generate path suggestions for the whole configuration tree of Juniper MX:
+
+```
+gnmic --file junos/conf --dir common prompt
+```
+
+#### Arista
+YANG repo: [aristanetworks/yang](https://github.com/aristanetworks/yang)
+
+Arista uses a subset of OpenConfig modules and does not provide IETF modules inside their repo. So make sure you have IETF models available so you can reference it, a `openconfig/public` is a good candidate.
+
+Clone the Arista YANG repo:
+
+```
+git clone https://github.com/aristanetworks/yang
+cd yang
+```
+
+Generate path suggestions for all Arista OpenConfig modules:
+
+```
+gnmic --file EOS-4.23.2F/openconfig/public/release/models \
+      --dir ~/public/third_party/ietf \
+      --exclude ietf-interfaces \
+      prompt
+```
+
+## Enumeration suggestions
+`gnmic` flags that can take pre-defined values (enumerations) will get suggestions as well. For example, no need to keep in mind which subscription modes are available, the prompt will hint you:
+
+![enum suggestion](https://gitlab.com/rdodin/pics/-/wikis/uploads/a2772c709d869d5efc299db451e3d4a9/image.png)
+
+## File-path completions
+Whenever a user needs to provide a file path in a prompt mode, the filepath suggestions will make the process interactive:
+
+<script id="asciicast-uJyTI4nnQ52lSpIw5Ec7INLe7" src="https://asciinema.org/a/uJyTI4nnQ52lSpIw5Ec7INLe7.js" async></script>
+
+## Known issues
```

Comment:
I would leave this out, this is both terminal and goprompt dependent

Reply:
removed in a0d15077bf6f8e91f99d53aa7f65599e3a72e450